cmd/nerdctl/info.go

```diff
@@ -70,27 +73,98 @@ func infoAction(cmd *cobra.Command, args []string) error {
 	}
 	defer cancel()
 
-	snapshotter, err := cmd.Flags().GetString("snapshotter")
+	mode, err := cmd.Flags().GetString("mode")
 	if err != nil {
 		return err
 	}
-	cgroupManager, err := cmd.Flags().GetString("cgroup-manager")
-	if err != nil {
-		return err
+	switch mode {
+	case "dockercompat", "native":
+	default:
+		return fmt.Errorf("unknown mode %q", mode)
 	}
```

Comment:
to avoid redondant switch case : 
```suggestion
	switch mode {
	case "native":
		di, err := infoutil.NativeDaemonInfo(ctx, client)
		if err != nil {
			return err
		}
		ci, err := nativeClientInfo(cmd)
		if err != nil {
			return err
		}
		infoNative = &native.Info{
			Client: ci,
			Daemon: di,
		}
	case "dockercompat":
		snapshotter, err := cmd.Flags().GetString("snapshotter")
		if err != nil {
			return err
		}
		cgroupManager, err := cmd.Flags().GetString("cgroup-manager")
		if err != nil {
			return err
		}
		infoCompat, err = infoutil.Info(ctx, client, snapshotter, cgroupManager)
		if err != nil {
			return err
		}
	default:
		return fmt.Errorf("unknown mode %q", mode)
	}
```

Reply:
fixed